db/database_collection.go

```diff
@@ -249,7 +249,7 @@ func (c *DatabaseCollection) syncAndUserXattrKeys() []string {
 
 // syncMouAndUserXattrKeys returns the xattr keys for the user, mou and sync xattrs.
 func (c *DatabaseCollection) syncMouAndUserXattrKeys() []string {
-	xattrKeys := []string{base.SyncXattrName, base.VvXattrName}
+	xattrKeys := []string{base.SyncXattrName, base.VvXattrName, base.DocumentXattrKey}
 	if c.useMou() {
 		xattrKeys = append(xattrKeys, base.MouXattrName)
 	}
```

Comment:
I think the intent behind this if statement was to have _only_ two xattrs in pre 7.6 when we can do multi subdoc operations https://github.com/couchbase/sync_gateway/blob/2f5dcd6585140f9dfb6d5fae8fc15d772f2e6ff6/base/collection_xattr.go#L196

It reads weirdly to me now. I think on anemone we require 7.6. If we don't check for 7.6 on startup I think that we should (Out of scope for this PR) and then we can drop the check inside this function.

Reply:
Just for correctness I have moved the virtual xattr inside this check.